dspace-server-webapp/src/test/java/org/dspace/app/rest/LDNInboxControllerIT.java

```diff
@@ -0,0 +1,332 @@
+                                .withUrl("service url")
+                                .withLdnUrl("https://overlay-journal.com/inbox/")
+                                .withLowerIp("127.0.0.1")
+                                .withUpperIp("127.0.0.3")
+                                .build();
+        context.restoreAuthSystemState();
+
+        InputStream announceEndorsementStream = getClass().getResourceAsStream("ldn_announce_endorsement.json");
+        String announceEndorsement = IOUtils.toString(announceEndorsementStream, Charset.defaultCharset());
+        announceEndorsementStream.close();
+        String message = announceEndorsement.replaceAll("<<object>>", object);
+        message = message.replaceAll("<<object_handle>>", object);
+
+        ObjectMapper mapper = new ObjectMapper();
+        Notification notification = mapper.readValue(message, Notification.class);
+        getClient()
+            .perform(post("/ldn/inbox")
+                .contentType("application/ld+json")
+                .content(message))
+            .andExpect(status().isAccepted());
+
+        LDNMessageEntity ldnMessage = ldnMessageService.find(context, notification.getId());
+        checkStoredLDNMessage(notification, ldnMessage, object);
+    }
+
+    @Test
+    public void ldnInboxAnnounceReviewTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context).withName("community").build();
+        Collection collection = CollectionBuilder.createCollection(context, community).build();
+        Item item = ItemBuilder.createItem(context, collection).build();
+        InputStream announceReviewStream = getClass().getResourceAsStream("ldn_announce_review.json");
+        String object = configurationService.getProperty("dspace.ui.url") + "/handle/" + item.getHandle();
+        NotifyServiceEntity notifyServiceEntity = NotifyServiceBuilder.createNotifyServiceBuilder(context)
+                                .withName("service name")
+                                .withDescription("service description")
+                                .withUrl("https://review-service.com/inbox/about/")
+                                .withLdnUrl("https://review-service.com/inbox/")
+                                .withScore(BigDecimal.valueOf(0.6d))
+                                .withLowerIp("127.0.0.1")
+                                .withUpperIp("127.0.0.3")
+                                .build();
+        String announceReview = IOUtils.toString(announceReviewStream, Charset.defaultCharset());
+        announceReviewStream.close();
+        String message = announceReview.replaceAll("<<object>>", object);
+        message = message.replaceAll("<<object_handle>>", object);
+
+        ObjectMapper mapper = new ObjectMapper();
+        Notification notification = mapper.readValue(message, Notification.class);
+        getClient()
+            .perform(post("/ldn/inbox")
```

Comment:
Hello @tdonohue the endpoint `/ldn/inbox` has no authentication/authorization feature since the endpoint has to be publicly available and no authorization feature is currently supported by external serivices.
The way we addressed the "authentication" is by listing **trusted** LDNServices each Notification is processed only if the service has been registered and is active. Also there's an IP range validation for any registered LDN Service which is preventing malicious services/bots to "impersonate" services by simply changing the LDN Service url in the LDN Message.

The process can be summarized in this way:
1. The system receives the notification and extracts the LDN Service URL
2. The service URL is used to lookup in the database for registered (and active) services
3. Once the LDN Service is retrieved the IP address of the request is checked against the IP range defined for the LDN Service (Notice the IP range is a mandatory field when submitting a new LDN service)

Please let us know if you have any concerns

Reply:
@steph-ieffam : Thanks for clarifying.  That sounds like it would be sufficient.  I must have overlooked in the code how these services are "registered" with DSpace.  But, as long as we only "accept" notifications from trusted services/IPs, then that seems good to me.

**That said** I would recommend we add ITs to this class to verify that an *unregistered* service or invalid IP will be *blocked or ignored* in some manner.  It's possible I overlooked those tests, but I didn't notice them in my initial review.

I'll revisit this when I find time to fully test everything, just to be sure the service registration is sufficient.